weaver/main.go

```diff
@@ -145,6 +145,11 @@ func handleHttp(router *weave.Router) {
 		io.WriteString(w, fmt.Sprintln("Encryption", encryption))
 		io.WriteString(w, router.Status())
 	})
+	http.HandleFunc("/status-json", func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("Access-Control-Allow-Origin", "*")
```

Comment:
This looks dodgy. It means any site can request this document (by guessing, e.g., a docker IP) and thereby get information on the weave network. Presently the HTTP endpoint is only exposed to the host network, but this way lies vulnerabilities.